Homework3

EremitePK opened the pull request:
Исправлены ошибки прошлого домашнего задания.
Реализовано последнее домашнее задание.
Переход в новый фрагмент возникает при нажатии на постер фильма. Также назад открывается список фильмов при нажатии на надпись "back".
![Screenshot_1606504412](https://user-images.githubusercontent.com/50176211/100479235-2d150980-3120-11eb-83f4-d9e338e25276.png)
![Screenshot_1606504417](https://user-images.githubusercontent.com/50176211/100479241-30a89080-3120-11eb-9556-373dcdb53e8f.png)

EremitePK committed:
Начата реализация 3 домашнего задания и исправлены коментарии по 2 домашнему заданию

EremitePK committed:
Домашняя работа 3.
Реализованы формы и переходы по нажатию на постер в первом окне и при нажатии на кнопку "back" во втором окне

EremitePK committed:
Домашняя работа 3.
Реализованы формы и переходы по нажатию на постер в первом окне и при нажатии на кнопку "back" во втором окне.
Исправлено открытие фрагмента со списком фильмов при повороте экрана с открытым фрагментом деталей фильма.

EremitePK committed:
Merge branches 'Homework3' and 'master' of https://github.com/EremitePK/MyApplication into Homework3

EremitePK committed:
Переделал взаимодействие форм в соответствии с домашним заданием

ivanlardis commented on app/src/main/java/ru/eremite/myapplication/MainActivity.kt:
```diff
@@ -1,38 +1,31 @@
-            Snackbar.make(view, "Replace with your own action", Snackbar.LENGTH_LONG)
-                    .setAction("Action", null).show()
+        if (savedInstanceState == null) {
+            OnMoviesListActiv()
```
метод с маленькой буквы

ivanlardis commented on app/src/main/java/ru/eremite/myapplication/MainActivity_old.kt:
```diff
@@ -0,0 +1,38 @@
+import android.view.Menu
+import android.view.MenuItem
+
+class MainActivity_old : AppCompatActivity() {
```
Давай удалять те активити которые не нужны в проекте)
сложно ревьювить и разбиратся что нужно что нет)

ivanlardis commented on app/src/main/java/ru/eremite/myapplication/MovieDetailsActivity.kt:
```diff
@@ -1,11 +1,41 @@
+import android.widget.TextView
+import androidx.fragment.app.Fragment
+
+class MovieDetailsActivity : Fragment() {
```
это фрагмент а не активити
название очень путает)

ivanlardis commented on app/src/main/java/ru/eremite/myapplication/MovieDetailsActivity.kt:
```diff
@@ -1,11 +1,41 @@
+
+    override fun onDetach() {
+        super.onDetach()
+        listener = null
```
писали в группе что сначала лучше
 listener = null 
а потом 
 super.onDetach()
Но это не критично)

ivanlardis commented on app/src/main/java/ru/eremite/myapplication/MainActivity.kt:
```diff
@@ -1,38 +1,31 @@
+    override fun OnMoviesListActiv() {
+        supportFragmentManager.beginTransaction()
+                ?.replace(R.id.fragment_container_view, FragmentMoviesList())
+                ?.commit()
```
создаем фрагметы через вызов
FragmentMoviesList.newInstance()
как в лекции

ivanlardis commented on app/src/main/java/ru/eremite/myapplication/MovieDetailsActivity.kt:
```diff
@@ -1,11 +1,41 @@
+    private var listener: TopMainMenuClickListener? = null
 
-class MovieDetailsActivity : AppCompatActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
```
если не используем метод то удаляем

ivanlardis commented on app/src/main/java/ru/eremite/myapplication/MovieDetailsActivity.kt:
```diff
@@ -1,11 +1,41 @@
+    override fun onDetach() {
+        super.onDetach()
+        listener = null
+    }
```
Добавить 
companion object {
        fun newInstance(): MoviesListFragment {
            return MoviesListFragment()
        } 
    }

Что бы создавать фрагменты через него

ivanlardis commented on app/src/main/java/ru/eremite/myapplication/FragmentMoviesList.kt:
```diff
@@ -0,0 +1,77 @@
+ * Use the [FragmentMoviesList.newInstance] factory method to
+ * create an instance of this fragment.
+ */
+class FragmentMoviesList : Fragment() {
```
Нужно сделать подобным фрагменту
который получится после исправления коментариев 
в файле MovieDetailsActivity

ivanlardis replied:
Непонятно почему сделано два фрагмента
И один из них более менее ок
А второй этот)

EremitePK committed:
Исправил по коментариям:
1. Удалил все не нужный активити и лаяты.
2. Реализовал вызов фрагментов через newIntens
3. Иправил наименования фрагментов.
4. Иправил код в фрагментах (до этого код создался по шаблону)
Вроде все учел и исправил.
Заранее извеняюсь за крайне не понятное и неправильное наисание всего в этом коде.

EremitePK commented:
Исправил по коментариям:
1. Удалил все не нужный активити и лаяты.
2. Реализовал вызов фрагментов через newIntens
3. Иправил наименования фрагментов.
4. Иправил код в фрагментах (до этого код создался по шаблону)
Вроде все учел и исправил.
Заранее извеняюсь за крайне не понятное и неправильное наисание всего в этом коде.
